src/ansys/openapi/common/_api_client.py

```diff
@@ -378,18 +381,20 @@ def __deserialize(self, data: SerializedType, klass_name: str) -> DeserializedTy
 
         if klass_name in self.NATIVE_TYPES_MAPPING:
             klass = self.NATIVE_TYPES_MAPPING[klass_name]
-        else:
-            klass = self.models[klass_name]
-
-        if klass in self.PRIMITIVE_TYPES:
-            assert isinstance(data, (str, int, float, bool, bytes))
-            return self.__deserialize_primitive(data, klass)
-        elif klass == datetime.date:
+            if klass in self.PRIMITIVE_TYPES:
+                assert isinstance(data, (str, int, float, bool, bytes))
+                return self.__deserialize_primitive(data, klass)
+            elif klass == datetime.date:
+                assert isinstance(data, str)
+                return self.__deserialize_date(data)
+            elif klass == datetime.datetime:
+                assert isinstance(data, str)
+                return self.__deserialize_datetime(data)
+
+        klass = self.models[klass_name]
+        if issubclass(klass, Enum):
             assert isinstance(data, str)
-            return self.__deserialize_date(data)
-        elif klass == datetime.datetime:
-            assert isinstance(data, str)
-            return self.__deserialize_datetime(data)
+            return klass(data)
```

Comment:
I had to nest the if statements to help mypy figure out what the type is in the last else clause. Without this, it was unable to figure out that we can only be left with a BaseModel